src/db.rs

```diff
@@ -43,388 +43,374 @@ impl Default for Connection {
     }
 }
 
-pub async fn get_dbpool() -> Result<AnyPool> {
-    let db_type = env::var("KBS_DB_TYPE").expect("KBS_DB_TYPE not set");
-    let host_name = env::var("KBS_DB_HOST").expect("KBS_DB_HOST not set");
-    let user_name = env::var("KBS_DB_USER").expect("KBS_DB_USER not set.");
-    let db_pw = env::var("KBS_DB_PW").expect("KBS_DB_PW not set.");
-    let data_base = env::var("KBS_DB").expect("KBS_DB not set");
-
-    let db_url = if db_type == "sqlite" {
-        format!("{}://{}", db_type, data_base)
-    } else {
-        format!(
-            "{}://{}:{}@{}/{}",
-            db_type, user_name, db_pw, host_name, data_base
-        )
-    };
-
-    let db_pool = AnyPoolOptions::new()
-        .max_connections(1000)
-        .connect(&db_url)
-        .await
-        .map_err(|e| {
-            anyhow!(
-                "db::get_db_pool:: Encountered error trying to create database pool: {}",
-                e
-            )
-        })?;
-    Ok(db_pool)
+pub struct KbsDb {
+    dbpool: AnyPool,
 }
 
-fn replace_binds(kind: AnyKind, sql: &str) -> String {
-    if kind != AnyKind::Postgres {
-        return sql.to_string();
+impl KbsDb {
+    pub async fn new() -> Result<Self> {
+        let db_type = env::var("KBS_DB_TYPE").expect("KBS_DB_TYPE not set");
+        let host_name = env::var("KBS_DB_HOST").expect("KBS_DB_HOST not set");
+        let user_name = env::var("KBS_DB_USER").expect("KBS_DB_USER not set.");
+        let db_pw = env::var("KBS_DB_PW").expect("KBS_DB_PW not set.");
+        let data_base = env::var("KBS_DB").expect("KBS_DB not set");
```

Comment:
Thanks for the review @jodh-intel -- I agree with your comments.

However, in this PR I tried to do only one thing (which is to avoid calling `get_dbpool` each time before a query is executed). If you look at this PR with "hide whitespace changes" turned on, you'll see these lines haven't changed at all (just indented 4 spaces deeper).

I think we should resolve the issues you raised (expect, unwrap, max-connections) in a separate PRs.

Reply:
I will make these changes after Dov's PR is approved.

Reply:
These are updated and should be in the next PR.